src/main/java/org/apache/sling/models/impl/FirstImplementationPicker.java

```diff
@@ -31,9 +29,7 @@
  * But at least it gives a consistent behavior.
  * It's service ranking is set to the highest value to allow more intelligent implementations to step in.
  */
-@Component
-@Service
-@Property(name = Constants.SERVICE_RANKING, intValue = Integer.MAX_VALUE)
+@Component(property=Constants.SERVICE_RANKING+":Integer="+Integer.MAX_VALUE)
```

Comment:
cosmetic: also not explicitly required i personally prefer to add an explicit service=ImplementationPicker.class attribute to make it clear that this is an OSGi service and not only a component. (same case may apply to other services as well)